File: src/main/java/org/sopt/common/response/CustomAPIResponse.java
package org.sopt.common.response;

public record CustomAPIResponse<T>(
	String status,
	String message,
	T data
) {
	// 성공
	public static <T> CustomAPIResponse<T> createSuccess(SuccessStatus status, T data) {
		return new CustomAPIResponse<T>(String.valueOf(status.getStatus().value()), status.getMessage(), data);
	}

	// 실패
	public static <T> CustomAPIResponse<T> createFail(String status, String message) {
		return new CustomAPIResponse<T>(status, message, null);
	}

	public static <T> CustomAPIResponse<T> createFail(String status, String message, T data) {
		return new CustomAPIResponse<>(status,message, data);
	}
}

File: src/main/java/org/sopt/common/response/CustomAPIResponse.kt
package org.sopt.common.response

data class CustomAPIResponse<T>(
    val status: String,
    val message: String,
    val data: T
) {
    companion object {
        // 성공
        fun <T> createSuccess(
            status: SuccessStatus,
            data: T
        ): CustomAPIResponse<T> =
            CustomAPIResponse(
                status.status.value().toString(),
                status.message,
                data
            )

        // 실패
        fun createFail(
            status: String,
            message: String
        ): CustomAPIResponse<Any?> =
            CustomAPIResponse(
                status,
                message,
                null
            )

        fun <T> createFail(
            status: String,
            message: String,
            data: T
        ): CustomAPIResponse<T> =
            CustomAPIResponse(
                status,
                message,
                data
            )
    }
}
